atlasdb-cassandra/src/test/java/com/palantir/atlasdb/cassandra/RefreshableWithInitialDefaultTest.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.cassandra;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.palantir.logsafe.exceptions.SafeIllegalArgumentException;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.refreshable.SettableRefreshable;
+import java.util.function.Function;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RefreshableWithInitialDefaultTest {
+
+    @Mock
+    Function<Integer, String> function;
+
+    @Test
+    public void preservesValidInitialValueIfFollowingValueIsNotValid() {
```

Comment:
There are some tests in `CassandraAtlasDbFactory` around this behaviour that haven't been broken, but the following tests are slightly more extensive.